### DISCUSSION TITLE
importing `hardhat` in `javascript` to `compile` solidity code

### DISCUSSION THREAD

**Discussion by hammersharkfish** [NomicFoundation/hardhat#4982] [General]
I want to use `hardhat`to `compile` in a script and not in bash is there any way to achieve this ?  
`
import hardhat from "hardhat" 
.
.
.
hardhat.compile(" pragma solidity ...")
`

**Comment by kanej** [comment]
You can run Hardhat's tasks programmatically from a script:

```js
// ./scripts/compile.js

// Import Hardhat's runtime environment
const hre = require("hardhat");

async function main() {
  // Run the compile task
  await hre.run("compile");

  console.log("Compilation finished successfully");
}

main()
  .then(() => process.exit(0))
  .catch((error) => {
    console.error(error);
    process.exit(1);
  });

```

You can then run the script with:

```sh
npx hardhat run ./scripts/compile.js
```

**Reply by hammersharkfish** [reply]
Thank you . I will try this and get back

**Comment by hammersharkfish** [comment]
This works when compiling local hardhat project (thank you for that) but when I try to point it outside the project directory it throws
`HardhatError: HH1007: The file 'some/path/outside/contracts',is treated as local but is outside the project` 
Here is my `path` of `hardhat.config.cjs` (of local directory)
```
  paths: {
    sources: 'some/path/outside/contracts',
    tests: ' some/path/outside/tests',
    cache: 'some/path/outside/cache',
    artifacts: 'some/path/outside/artifacts',
  }
```
My understanding is `paths` is used to find `sources` from any directory so why this error .
